examples/fastaiv2_simple.py

```diff
@@ -0,0 +1,108 @@
+"""
+Optuna example that optimizes convolutional neural network and data augmentation using FastAI V2.
+
+In this example, we optimize the hyperparameters of a convolutional neural network and
+data augmentation for hand-written digit recognition in terms of validation accuracy.
+The network is implemented by fastai and
+evaluated on MNIST dataset. Throughout the training of neural networks, a pruner observes
+intermediate results and stops unpromising trials.
+Note that this example will take longer than the other examples
+as this uses the entire MNIST dataset.
+
+You can run this example as follows, pruning can be turned on and off with the `--pruning`
+argument.
+    $ python fastaiv2_simple.py [--pruning]
+"""
+
+import argparse
+
+from fastai.vision.all import accuracy
+from fastai.vision.all import aug_transforms
+from fastai.vision.all import CudaCallback
+from fastai.vision.all import ImageDataLoaders
+from fastai.vision.all import Learner
+from fastai.vision.all import SimpleCNN
+from fastai.vision.all import untar_data
+from fastai.vision.all import URLs
+
+import optuna
+from optuna.integration import FastAIPruningCallback
+
+
+BATCHSIZE = 128
+EPOCHS = 10
+
+
+path = untar_data(URLs.MNIST_SAMPLE)
+
+
+def objective(trial):
+    # Data Augmentation
+    apply_tfms = trial.suggest_categorical("apply_tfms", [True, False])
+    if apply_tfms:
+        # MNIST is a hand-written digit dataset. Thus horizontal and vertical flipping are
+        # disabled. However, the two flipping will be important when the dataset is CIFAR or
+        # ImageNet.
+        tfms = aug_transforms(
+            do_flip=False,
+            flip_vert=False,
+            max_rotate=trial.suggest_int("max_rotate", 0, 45),
+            max_zoom=trial.suggest_float("max_zoom", 1, 2),
+            p_affine=trial.suggest_discrete_uniform("p_affine", 0.1, 1.0, 0.1),
+        )
+    data = ImageDataLoaders.from_folder(
+        path, bs=BATCHSIZE, batch_tfms=tfms if apply_tfms else None
+    )
+
+    n_layers = trial.suggest_int("n_layers", 2, 5)
+
+    n_channels = [3]
+    for i in range(n_layers):
+        out_channels = trial.suggest_int("n_channels_{}".format(i), 3, 32)
+        n_channels.append(out_channels)
+    n_channels.append(2)
+
+    model = SimpleCNN(n_channels)
+
+    learn = Learner(
+        data,
+        model,
+        metrics=[accuracy],
+        # You could as FastAIPruningCallback in the fit function
+        cbs=[FastAIPruningCallback(trial), CudaCallback],
```

Comment:
Ditto. `CudaCallback` doesn't seem to be necessary.

Reply:
Here, you are right. It isn't necessary.